src/Doctrine/Orm/State/LinksHandlerTrait.php

```diff
@@ -20,12 +20,19 @@
 use Doctrine\ORM\Mapping\ClassMetadataInfo;
 use Doctrine\ORM\QueryBuilder;
 
+/**
+ * @internal
+ */
 trait LinksHandlerTrait
 {
     use CommonLinksHandlerTrait;
 
-    private function handleLinks(QueryBuilder $queryBuilder, array $identifiers, QueryNameGenerator $queryNameGenerator, array $context, string $entityClass, Operation $operation): void
+    private function handleLinks(QueryBuilder $queryBuilder, array $identifiers, QueryNameGenerator $queryNameGenerator, array $context, string $entityClass, Operation $operation = null): void
```

Comment:
If `$operation` can be null, you should change the line 44 `$links = $this->getLinks($entityClass, $operation, $context);`